### ISSUE TITLE
[Flutter][Android] Not working

### ISSUE THREAD

**Issue by vanvixi** [usefulness/easylauncher-gradle-plugin#481]
I am configuring easylauncher for my application and I have not succeeded

Here is my configuration
**build.gradle**

```
buildscript {
    ext.kotlin_version = '1.7.10'
    repositories {
        google()
        mavenCentral()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:7.3.0'
        // START: FlutterFire Configuration
        classpath 'com.google.gms:google-services:4.3.10'
        // END: FlutterFire Configuration
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"

    }
}

plugins {
    id 'com.starter.easylauncher' version '5.1.2' apply false
}
allprojects {
    repositories {
        google()
        mavenCentral()
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

tasks.register("clean", Delete) {
    delete rootProject.buildDir
}
```

**app/build.gradle**

```
def localProperties = new Properties()
def localPropertiesFile = rootProject.file('local.properties')
if (localPropertiesFile.exists()) {
    localPropertiesFile.withReader('UTF-8') { reader ->
        localProperties.load(reader)
    }
}

def keystoreProperties = new Properties()
def keystorePropertiesFile = rootProject.file('key.properties')
if (keystorePropertiesFile.exists()) {
    keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
}

def flutterRoot = localProperties.getProperty('flutter.sdk')
if (flutterRoot == null) {
    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
}

def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
if (flutterVersionCode == null) {
    flutterVersionCode = '1'
}

def flutterVersionName = localProperties.getProperty('flutter.versionName')
if (flutterVersionName == null) {
    flutterVersionName = '1.0'
}

def flutterMinSdkVersion = localProperties.getProperty('flutter.minSdkVersion')
if (flutterMinSdkVersion == null) {
    flutterMinSdkVersion = '21'
}

def flutterNdkVersion = localProperties.getProperty('flutter.ndkVersion')
if (flutterNdkVersion == null) {
    flutterNdkVersion = '25.1.8937393'
}

apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'

apply plugin: 'com.starter.easylauncher'

apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"

android {
    /// ----- BEGIN SETUP Flavor -----
    flavorDimensions "flavorType"

    productFlavors {
        local {
            dimension "flavorType"
            applicationIdSuffix ".local"
            resValue "string", "app_name", "App(local)"
        }
        dev {
            dimension "flavorType"
            applicationIdSuffix ".dev"
            resValue "string", "app_name", "App(dev)"
        }
        stg {
            dimension "flavorType"
            applicationIdSuffix ".stg"
            resValue "string", "app_name", "App(stg)"
        }
        prod {
            dimension "flavorType"
            resValue "string", "app_name", "App"
        }
    }
    /// ----- END SETUP Flavor -----

    namespace "com.ex.app"
    compileSdkVersion flutter.compileSdkVersion
    ndkVersion flutterNdkVersion

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    packagingOptions {
        pickFirst 'lib/armeabi-v7a/libc++_shared.so'
        pickFirst 'lib/arm64-v8a/libc++_shared.so'
        pickFirst 'lib/x86/libc++_shared.so'
        pickFirst 'lib/x86_64/libc++_shared.so'
    }

    kotlinOptions {
        jvmTarget = '1.8'
    }

    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId "com.ex.app"
        // You can update the following values to match your application needs.
        // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-gradle-build-configuration.
        minSdkVersion flutterMinSdkVersion
        targetSdkVersion flutter.targetSdkVersion
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
        multiDexEnabled = true
    }

    signingConfigs {
        release {
            keyAlias keystoreProperties['keyAlias']
            keyPassword keystoreProperties['keyPassword']
            storeFile keystoreProperties['storeFile'] ? file(keystoreProperties['storeFile']) : null
            storePassword keystoreProperties['storePassword']
        }
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.release
        }
    }
}

easylauncher {
    iconNames = ["@mipmap/ic_launcher"]
    defaultFlavorNaming(true)

    buildTypes {
        debug {}
        release {}
    }

    productFlavors {
        local {
            filters = orangeRibbonFilter()
        }
        dev {
            filters = redRibbonFilter()
        }
        stg {
            filters = blueRibbonFilter()
        }
        prod {
            enable false  // No labels in production
        }
    }
}

flutter {
    source '../..'
}

dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation("androidx.multidex:multidex:2.0.1")
    implementation 'com.facebook.android:facebook-android-sdk:[8,9)'
}
```

**Res**

<img width="412" alt="Screenshot 2023-08-02 at 11 54 39" src="https://github.com/usefulness/easylauncher-gradle-plugin/assets/75975945/1863d2a6-b61d-4021-a3f2-9bd6ca187599">

**Comment by vanvixi** [comment]
@mateuszkwiecinski Can you help me?

**Comment by mateuszkwiecinski** [comment]
I'm sorry, I can't help with debugging your setup. Please do provide a project reproducing the issue. 

Scanning through the code you pasted, it looks like you're using very old version `5.1.2`, which may not support recent AGP versions, but that's just a blind guess

**Comment by vanvixi** [comment]
- This is my example project.
- I am having problem adding easylauncher to my app for versions higher than 5.1.2

https://github.com/vanvixi/flutter_easylauncher/tree/main/android

**Comment by mateuszkwiecinski** [comment]
>I am having problem adding easylauncher to my app for versions higher than 5.1.2

You have to use latest easylaucher version to make the project work with AGP 7.3 and up. Or use lower AGP version 🤷 

>This is my example project.

I had trouble to make the project compile, but after commenting bits and bobs and upgrading to latest AGP & easylauncher version result in the ribbon being shown. There is nothing more the plugin can do. You have to fix your project and use latest versions. AGP between 7.3 and 7.4 was utterly broken, so I won't be surprised if things don't work there

**Comment by vanvixi** [comment]
Thanks. 
The error was fixed after I updated AGP 7.4.2
